src/frontend.cs

```diff
@@ -1582,6 +1576,10 @@ public override object VisitReturn(bhlParser.ReturnContext ctx)
     }
     else
     {
+      var call_post_inc_mixed = ctx.expCallPostIncMixedList();
+      if(call_post_inc_mixed != null)
+        FireError(Location(ctx) + ": operator ++ is not allowed a return");
```

Comment:
This looks a bit suspicious... Can't we rather handle it somehow during parsing by grammar rules? So that ANTLR parser would exit with an error?

Reply:
Or maybe this should be resolved by a more generic type check: expressions of void type can't be returned